src/main/java/org/jabref/gui/util/ViewModelTreeTableCellFactory.java

```diff
@@ -19,12 +20,12 @@
  */
 public class ViewModelTreeTableCellFactory<S, T> implements Callback<TreeTableColumn<S, T>, TreeTableCell<S, T>> {
 
-    private Callback<S, String> toText;
+    private Callback<S, StringProperty> toText;
     private Callback<S, Node> toGraphic;
     private Callback<S, EventHandler<? super MouseEvent>> toOnMouseClickedEvent;
     private Callback<S, String> toTooltip;
 
-    public ViewModelTreeTableCellFactory<S, T> withText(Callback<S, String> toText) {
```

Comment:
Changing this to `StringProperty` shouldn't be necessary. Normally, the table automatically refreshes the cell if the underlying data is changed. Please revert this change even if then the bug is no longer fixed (see the general comment).